benchmarks/include/map_parameters/map_parameters.h

```diff
@@ -0,0 +1,167 @@
+        for (int i = 0; i < threadCount; i++) {
+            vectorThreads[i].join();
+        }
+        auto it = std::max_element(std::begin(times), std::end(times));
+        return *it;
+    }
+
+    template<typename Functor, typename Functor2>
+    double measureOperationMultithreaded(const Functor& operationWrite, 
+    const Functor2& operationRead, 
+    std::function<V(int)> valueCreator) {
+        std::vector<std::thread> vectorThreads;
+        std::vector<double> times(threadCount);
+        std::atomic<int> barrier(0);
+        vectorThreads.emplace_back([&](int id) {
+                barrier++;
+                while (barrier < threadCount + 1) {
+                    std::this_thread::sleep_for(std::chrono::milliseconds(1));
+                }
+                times[id] = measureOperation(operationWrite, keys[0], valueCreator);
+
+        }, 0);
+        for (int i = 1; i < threadCount; i++) {
+            vectorThreads.emplace_back([&](int id) {
+                barrier++;
+                while (barrier < threadCount + 1) {
+                    std::this_thread::sleep_for(std::chrono::milliseconds(1));
+                }
+                times[id] = measureOperation(operationRead, keys[i]);
+
+            }, i);
+        }
+        while (barrier < threadCount ) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        }
+        barrier++;
+        for (int i = 0; i < threadCount; i++) {
+            vectorThreads[i].join();
+        }
+        auto it = std::max_element(std::begin(times), std::end(times));
+        return *it;
+    }
+    template<typename Functor>
+    double measureOperation(Functor operation) {
+        auto start = std::chrono::high_resolution_clock::now();
+        for (int i = 0; i < mapSize; i++) {
+            operation(keys[i]);
+        }
+        auto end = std::chrono::high_resolution_clock::now();
+        auto elapsed_seconds = std::chrono::duration_cast<std::chrono::duration<double>>(end - start);
+        return elapsed_seconds.count();
+    }
+    template<typename Functor>
+    double measureOperation(Functor operation, std::function<V(int)> valueCreator) {
+        auto start = std::chrono::high_resolution_clock::now();
+        for (int i = 0; i < mapSize; i++) {
+            operation(keys[i], valueCreator(i));
+        }
+        auto end = std::chrono::high_resolution_clock::now();
+        auto elapsed_seconds = std::chrono::duration_cast<std::chrono::duration<double>>(end - start);
+        return elapsed_seconds.count();
+    }
+    int threadCount;
+    int mapSize;
+    std::function<std::string(int)> keyStringCreator;
+    std::function<unsigned(int)> keyUintCreator;
+    std::function<int(int)> valueIntCreator;
+    std::function<std::vector<int>(int)> valueVectorIntCreator;
+private:
+    template<typename Functor>
+    double measureOperation(const Functor& operation, K key) {
+        auto start = std::chrono::high_resolution_clock::now();
```

Comment:
Not sure if it worth using some wrapper like [this](https://stackoverflow.com/a/31392034) (see edited answer) so that we can easily reuse the code to get different granularity of results (seconds/microseconds/milliseconds etc).

Reply:
The correct way I think would be to add `->Unit(benchmark::kMillisecond)`  (with the correct time unit) to each benchmark registration of to map_parameters.cpp.

the method SetIterationTime of State assumes seconds.